components/org.wso2.carbon.identity.oauth.scope.endpoint/src/test/java/org/wso2/carbon/identity/oauth/scope/endpoint/util/ScopeUtilsTest.java

```diff
@@ -0,0 +1,121 @@
+
+        when(scopeDTO.getBindings()).thenReturn(scope);
+        when(scopeDTO.getName()).thenReturn("Lifecycle exception occurred");
+        when(scopeDTO.getDescription()).thenReturn("Error occurred while changing lifecycle state");
+
+        Scope scopeexpected = new Scope("Error occurred while changing lifecycle state", "Lifecycle exception occurred", new ArrayList<String>(Arrays.asList("scope1", "scope2")));
+        scopeexpected.setName("Lifecycle exception occurred");
+        scopeexpected.setDescription("Error occurred while changing lifecycle state");
+        scopeexpected.setBindings(scope);
+
+        Scope scope1 = ScopeUtils.getScope(scopeDTO);
+        Assert.assertEquals(scope1.getName(), scopeexpected.getName());
+        Assert.assertEquals(scope1.getDescription(), scopeexpected.getDescription());
+        Assert.assertEquals(scope1.getBindings(), scopeexpected.getBindings());
+
+    }
+
+    @Test(description = "Testing getScopeDTO")
+    public void testGetScopeDTO() throws Exception {
+
+        Scope scope = mock(Scope.class);
+        when(scope.getBindings()).thenReturn(new ArrayList<String>());
+        when(scope.getDescription()).thenReturn("Lifecycle exception occurred");
+        when(scope.getName()).thenReturn("Error occurred while changing lifecycle state");
+
+        ScopeDTO scopeDTOexpected = new ScopeDTO();
+        scopeDTOexpected.setBindings(new ArrayList<String>());
+        scopeDTOexpected.setDescription("Lifecycle exception occurred");
+        scopeDTOexpected.setName("Error occurred while changing lifecycle state");
+
+        ScopeDTO scopeDTO1 = ScopeUtils.getScopeDTO(scope);
+        Assert.assertEquals(scopeDTO1.getBindings(), scopeDTOexpected.getBindings());
+        Assert.assertEquals(scopeDTO1.getDescription(), scopeDTOexpected.getDescription());
+        Assert.assertEquals(scopeDTO1.getName(), scopeDTOexpected.getName());
+
+    }
+
+    @Test(description = "Testing getUpdateScope")
+    public void testGetUpdatedScope() throws Exception {
+
+        ScopeToUpdateDTO scopeToUpdateDTO = mock(ScopeToUpdateDTO.class);
+        ArrayList<String> scope = new ArrayList<>(Arrays.asList("scope1", "scope2"));
+        when(scopeToUpdateDTO.getDescription()).thenReturn("Error occurred while changing lifecycle state");
+        when(scopeToUpdateDTO.getBindings()).thenReturn(scope);
+
+        Scope scopeexpected = new Scope("openid", "Error occurred while changing lifecycle state", scope);
+        scopeexpected.setDescription("Error occurred while changing lifecycle state");
+        scopeexpected.setBindings(scope);
+        ScopeToUpdateDTO sc = new ScopeToUpdateDTO();
+
+        Scope scope1 = ScopeUtils.getUpdatedScope(scopeToUpdateDTO, "openid");
+        Assert.assertEquals(scope1.getDescription(), scopeexpected.getDescription());
+        Assert.assertEquals(scope1.getBindings(), scopeexpected.getBindings());
+
+    }
+
+    @Test
+    public void testGetScopeDTOs() throws Exception {
```

Comment:
Shall we remove this if this is not implemented ? Otherwise it will also be counted as a test.